File: backend/src/schemas/invoice.py
from pydantic import BaseModel, Field
from uuid import UUID
from datetime import datetime, date


class InvoiceCreate(BaseModel):
    student_id: UUID
    amount: float = Field(..., gt=0)
    description: str | None = None
    paid: bool = False
    payment_date: date | None = None
    signature: str | None = None


class InvoiceUpdate(BaseModel):
    amount: float | None = Field(None, gt=0)
    description: str | None = None
    paid: bool | None = None
    payment_date: date | None = None
    signature: str | None = None


class InvoiceResponse(BaseModel):
    id: UUID
    teacher_id: UUID
    student_id: UUID
    amount: float
    description: str | None = None
    paid: bool
    payment_date: date | None = None
    signature: str | None = None
    created_at: datetime

    model_config = {"from_attributes": True}

File: desktop_app/lib/core/models/invoice.dart
class InvoiceModel {
  final String id;
  final String studentId;
  final double amount;
  final String? description;
  final bool paid;
  final String? paymentDate;
  final String? signature;
  final DateTime createdAt;

  const InvoiceModel({
    required this.id,
    required this.studentId,
    required this.amount,
    this.description,
    required this.paid,
    this.paymentDate,
    this.signature,
    required this.createdAt,
  });

  factory InvoiceModel.fromJson(Map<String, dynamic> json) {
    return InvoiceModel(
      id: json['id']?.toString() ?? '',
      studentId: json['student_id']?.toString() ?? '',
      amount: (json['amount'] ?? 0).toDouble(),
      description: json['description']?.toString(),
      paid: json['paid'] ?? false,
      paymentDate: json['payment_date']?.toString(),
      signature: json['signature']?.toString(),
      createdAt: DateTime.tryParse(json['created_at']?.toString() ?? '') ?? DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'student_id': studentId,
        'amount': amount,
        'description': description,
        'paid': paid,
        'payment_date': paymentDate,
        'signature': signature,
        'created_at': createdAt.toIso8601String(),
      };
}
